typescript/nomad-deploy/src/bridge/index.ts

```diff
@@ -350,6 +353,104 @@ export async function enrollBridgeRouter(
   );
 }
 
+/**
+ * Deploy any preconfigured custom tokens. These tokens should be specified by
+ * ID in the configuration block
+ *
+ * @param local - The deploy instance for the chain on which to deploy custom
+ *  tokens
+ */
+export async function deployCustoms(local: AnyBridgeDeploy): Promise<void> {
+  if (!local.contracts.customs) local.contracts.customs = [];
+
+  // skip altogether if no customs in the config
+  const customs = local.config.customs;
+  if (!customs) return;
+
+  // preconditions
+  const implementationAddress =
+    local.contracts.bridgeToken?.implementation.address;
+  if (!implementationAddress)
+    throw new Error('need bridge token implementation');
+  if (!local.contracts.bridgeRouter)
+    throw new Error('need bridge router deployed');
+
+  // factories
+  const ubcFactory = new coreContracts.UpgradeBeaconController__factory(
+    local.deployer,
+  );
+  const beaconFactory = new coreContracts.UpgradeBeacon__factory(
+    local.deployer,
+  );
+  const proxyFactory = new coreContracts.UpgradeBeaconProxy__factory(
+    local.deployer,
+  );
+
+  for (const custom of customs) {
+    // deploy a custom controller
+    const controller = await ubcFactory.deploy(local.overrides);
+    await controller.deployTransaction.wait(local.chain.confirmations);
+
+    // deploy a custom beacon
+    const beacon = await beaconFactory.deploy(
+      implementationAddress,
+      controller.address,
+      local.overrides,
+    );
+    await beacon.deployTransaction.wait(local.chain.confirmations);
+
+    // deploy a proxy
+    const proxy = await proxyFactory.deploy(
+      beacon.address,
+      '0x',
+      local.overrides,
+    );
+    await proxy.deployTransaction.wait(local.chain.confirmations);
+
+    // pre-emptively transfer ownership of controller to governance
+    const relinquish = await controller.transferOwnership(
+      local.coreContractAddresses.governance.proxy,
+      local.overrides,
+    );
+    await relinquish.wait(local.chain.confirmations);
+
+    const tokenProxy = xAppContracts.BridgeToken__factory.connect(
+      proxy.address,
+      local.deployer,
+    );
+    // set initial details
+    await (
+      await tokenProxy.setDetails(custom.name, custom.symbol, custom.decimals)
+    ).wait(local.chain.confirmations);
+
+    // transfer ownership to the bridge router
+    await (
+      await tokenProxy.transferOwnership(
+        local.contracts.bridgeRouter.proxy.address,
+      )
+    ).wait(local.chain.confirmations);
+
+    // enroll the custom representation
```

Comment:
we should be cognizant of the fact that this will mean this token never emits a `TokenDeployed` event on the chain (this is fine just need to keep in mind..)

Reply:
for example this token will be overlooked in our script uploading tokens to Google Sheets